### PR TITLE
Core/AreaTriggers: Teleport to set instance entrances when entering i…

### PR DESCRIPTION
**Pull request by mdX7**
…nstances via areatrigger entity

**Tests performed:**
built, tested ingame

### CODE REVIEW

File: src/server/game/Entities/AreaTrigger/AreaTrigger.cpp [hunk 1]
```diff
@@ -756,8 +756,14 @@ void AreaTrigger::DoActions(Unit* unit)
                     case AREATRIGGER_ACTION_TELEPORT:
                     {
                         if (WorldSafeLocsEntry const* safeLoc = sObjectMgr->GetWorldSafeLoc(action.Param))
+                        {
                             if (Player* player = caster->ToPlayer())
+                            {
+                                if (WorldSafeLocsEntry const* instanceEntrance = player->GetInstanceEntrance(safeLoc->Loc.GetMapId()))
```
Comment (Shauren) [inline src/server/game/Entities/AreaTrigger/AreaTrigger.cpp:762]: So this makes having teleporting areatriggers impossible to use inside instances with all of them pointing to the entrance location now?

Comment (mdX7) [reply src/server/game/Entities/AreaTrigger/AreaTrigger.cpp:762]: fixed

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -19538,6 +19538,36 @@ void Player::AddInstanceEnterTime(uint32 instanceId, time_t enterTime)
         _instanceResetTimes.insert(InstanceTimeMap::value_type(instanceId, enterTime + HOUR));
 }
 
+WorldSafeLocsEntry const* Player::GetInstanceEntrance(uint32 targetMapId)
+{
+    if (GetMapId() == targetMapId)
+        return nullptr;
```
Comment (columbus-eristr) [inline src/server/game/Entities/Player/Player.cpp:19544]: Fetching the instance entrance of a player that is already inside the instance doesn't seem entirely unreasonable, feel like this scenario should be supported.

Comment (Krudor) [inline src/server/game/Entities/Player/Player.cpp:19544]: Seems reasonable to support fetching current entrance while inside the same map, any specific reason to why this should be prevented?

Comment (mdX7) [reply src/server/game/Entities/Player/Player.cpp:19544]: agreed

**Pull request merged**